src/codecs/md/plugins/cite.ts

```diff
@@ -2,90 +2,137 @@
+ *          a negative lookahead for this and need to deal with in plugin logic
+ * Group 3: Optional citation `suffix`
+ *
+ * See https://regex101.com/r/G6zvyw/2
  */
-const CITE_REGEX = /\B@([\w|-]+)(\.\w+)?/
-
-const locator: Locator = (value, fromIndex) => {
-  let index = -1
-  const found = []
-  index = value.indexOf(marker, fromIndex)
-  if (index !== -1) {
-    found.push(index)
-  }
-
-  if (!A.isEmpty(found)) {
-    found.sort((a, b) => a - b)
-    return found[0]
-  }
-
-  return -1
-}
+const NARRATIVE_REGEX = /@([\w-]+)(\.\w+)?(\s*\[([^\]]+)\])?/
+
+/* Regex to find Pandoc style parenthetical citations (in square brackets)
+ *
+ * Group 1: Optional prefix
+ * Group 2: Citation `target` id
+ * Group 3: Optional suffix (if starts with period instead of space assumed to be email)
+ * Group 4-: Possibly additional citations separated by colons
+ *
+ * See https://regex101.com/r/cPaCmO/1/
+ */
+const PARENTHETICAL_REGEX = /►(.*?)@([\w-]+)(.*?)((\s*;\s*.*?@[\w-]+.*?)*)?◄/
 
 export const citePlugin: Plugin<[]> = function () {
   const inlineTokenizer: Tokenizer = function (
     this: Parser,
     eat: Eat,
     value: string
   ) {
-    // allow escaping of all markers
+    // Allow escaping of markers
     // @ts-expect-error
     // eslint-disable-next-line @typescript-eslint/strict-boolean-expressions
-    if (!this.escape.includes(marker)) this.escape.push(marker)
+    if (!this.escape.includes('@')) this.escape.push('@')
 
-    const startChar = value[0]
-    // @ts-expect-error
-    const now = eat.now()
-    now.column += 1
-    now.offset += 1
-
-    if (
-      startChar === marker &&
-      !value.startsWith(marker + ' ') &&
-      !value.startsWith(marker + marker)
-    ) {
-      const matches = CITE_REGEX.exec(value) ?? []
+    // Parenthetical citations
+    if (value[0] === '►') {
+      const match = PARENTHETICAL_REGEX.exec(value)
+
+      // Early termination if we don’t have a match
+      if (!match) return
 
-      const isEmail = matches[2] !== undefined
+      const [_all, prefix, target, suffix, more] = match
 
-      // Early termination if we’re dealing with an email and not a citation format
-      if (isEmail) return
+      // Early termination if we’re dealing with an email link
+      if (
+        more !== undefined &&
+        /\w$/.test(prefix) &&
+        /\.[a-z]{2,}/.test(suffix)
+      )
+        return
+
+      let groups = [[prefix, target, suffix]]
+      if (more !== undefined) {
+        groups = [
+          ...groups,
+          ...more
+            .split(/\s*;\s*/)
+            .slice(1)
+            .map((cite) => (/^(.*?)@([\w-]+)(.*?)$/.exec(cite) ?? []).slice(1)),
+        ]
+      }
+
+      // Handle potentially more than one Cite
+      const items = groups.map(([prefix, target, suffix]) => {
+        if (prefix !== undefined) prefix = prefix.trim()
+        if (suffix !== undefined) suffix = suffix.trim()
```

Comment:
Can you give me some context on the need to `trim` these please?
Reason I ask is because this causes issues when going to other formats.
`[e.g. @ref1; @ref2 and others]` to html results in:
![Screenshot 2021-03-19 at 12 03 23](https://user-images.githubusercontent.com/1646307/111810054-f9ffad00-88ab-11eb-8819-3181ddfaf638.png)

Do you think something like this would be better?
```suggestion
        if (prefix !== undefined) prefix = prefix.trim() + ' '
        if (suffix !== undefined) suffix = ' ' + suffix.trim()
```

Reply:
The reason I trimmed was because the spaces after a prefix and before a suffix are not really part of the prefix or suffix but are rather separators from the `target`. 

Pandoc does the same for prefixes, but for suffixes "trims" the preceding white space to a single space (not sure why):

```
> echo '[prefix @ref1    suffix  ]' | pandoc --from markdown --to json | jq .blocks[0].c[0].c[0]
[
  {
    "citationId": "ref1",
    "citationPrefix": [
      {
        "t": "Str",
        "c": "prefix"
      }
    ],
    "citationSuffix": [
      {
        "t": "Space"
      },
      {
        "t": "Str",
        "c": "suffix"
      }
    ],
    "citationMode": {
      "t": "NormalCitation"
    },
    "citationNoteNum": 1,
    "citationHash": 0
  }
]
```

Currently separators are "added" to HTML via CSS in themes (as we do with separators between say authors and year in a `Cite`).

In Markdown, we add single separator spaces in: https://github.com/stencila/encoda/blob/926e49821c3a028478e1a34b3954823be5addfd7/src/codecs/md/index.ts#L1373

I'd prefer to leave as is.

Reply:
Thanks for clarifying. Given that we insert the whitespace in the Markdown codec, I'd prefer to avoid whitespace insertion in CSS unless we have to—as in the case of citations where we change element ordering.
Reason being that this causes issues when copy-pasting text, and we have already received a bug report for as much from a user.